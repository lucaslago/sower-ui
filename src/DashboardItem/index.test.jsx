import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import DashBoardItem from './index';

describe('<DashBoardItem />', () => {
  const props = {
    simulationService: {
      start: jest.fn().mockReturnValue(Promise.resolve({ message: 'ok' })),
      stop: jest.fn().mockReturnValue(Promise.resolve({ message: 'ok' })),
      create: jest.fn().mockReturnValue(Promise.resolve({ message: 'ok' })),
      status: jest.fn().mockReturnValue(Promise.resolve('fake status response')),
    },
    authToken: 'basic 123123',
    device: {
      id: '123',
      relationships: {
        equipment: {
          data: {
            description: 'a cool tractor simulation',
            default_simulation: true,
            id: 'anotherid',
            type: 'equipment',
            simulation_type: 'applicator',
          },
        },
      },
      simulationStatus: {
        status: 'inactive',
      },
    },
  };

  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <MuiThemeProvider>
        <DashBoardItem {...props} />
      </MuiThemeProvider>,
    );
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should expand <Card /> when start button is clicked', () => {
    expect(wrapper.find('.dashboard-item-progress-bar').length).toEqual(0);
    wrapper.find('.start button').simulate('click');
    setTimeout(() => {
      expect(wrapper.find('.dashboard-item-progress-bar').length).toEqual(1);
    }, 0);
  });

  it('should disable the start button after it is clicked', () => {
    const startButton = wrapper.find('.start button');
    expect(startButton.prop('disabled')).toEqual(false);
    startButton.simulate('click');
    expect(startButton.prop('disabled')).toEqual(true);
  });

  it('should enable the stop button after clicking the start button', () => {
    const startButton = wrapper.find('.start button');
    const stopButton = wrapper.find('.stop button');
    expect(stopButton.prop('disabled')).toEqual(true);
    startButton.simulate('click');
    setTimeout(() => {
      expect(stopButton.prop('disabled')).toEqual(false);
    }, 0);
  });
});
