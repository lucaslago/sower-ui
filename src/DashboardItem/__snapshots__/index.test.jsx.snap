exports[`<DashBoardItem /> renders correctly 1`] = `
<MuiThemeProvider>
  <DashboardItem
    authService={
      Object {
        "getToken": [Function],
      }
    }
    expanded={false}
    simulationService={
      Object {
        "start": [Function],
        "stop": [Function],
      }
    }
    startDisabled={false}
    stopDisabled={true}
    title="Tractor"
    trackerId="123">
    <Card
      className="DashboardItem"
      expandable={false}
      expanded={false}
      initiallyExpanded={false}
      style={
        Object {
          "marginTop": "1rem",
        }
      }>
      <Paper
        circle={false}
        className="DashboardItem"
        rounded={true}
        style={
          Object {
            "marginTop": "1rem",
            "zIndex": 1,
          }
        }
        transitionEnabled={true}
        zDepth={1}>
        <div
          className="DashboardItem"
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "#ffffff",
              "borderRadius": "2px",
              "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                     0 1px 4px rgba(0, 0, 0, 0.12)",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "Roboto, sans-serif",
              "marginTop": "1rem",
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "zIndex": 1,
            }
          }>
          <div
            style={
              Object {
                "paddingBottom": 0,
              }
            }>
            <CardHeader
              actAsExpander={false}
              avatar="http://icons.iconarchive.com/icons/elegantthemes/beautiful-flat-one-color/128/tractor-icon.png"
              showExpandableButton={false}
              subtitle="123"
              title="Tractor">
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontWeight": 500,
                    "muiPrepared": true,
                    "padding": 16,
                    "position": "relative",
                    "whiteSpace": "nowrap",
                  }
                }>
                <Avatar
                  size={40}
                  src="http://icons.iconarchive.com/icons/elegantthemes/beautiful-flat-one-color/128/tractor-icon.png"
                  style={
                    Object {
                      "marginRight": 16,
                    }
                  }>
                  <img
                    size={40}
                    src="http://icons.iconarchive.com/icons/elegantthemes/beautiful-flat-one-color/128/tractor-icon.png"
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgb(188, 188, 188)",
                        "borderRadius": "50%",
                        "color": "#ffffff",
                        "display": "inline-flex",
                        "fontSize": 20,
                        "height": 40,
                        "justifyContent": "center",
                        "marginRight": 16,
                        "muiPrepared": true,
                        "userSelect": "none",
                        "width": 40,
                      }
                    } />
                </Avatar>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "muiPrepared": true,
                      "paddingRight": "90px",
                      "verticalAlign": "top",
                      "whiteSpace": "normal",
                    }
                  }>
                  <span
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "display": "block",
                        "fontSize": 15,
                        "muiPrepared": true,
                      }
                    }>
                    Tractor
                  </span>
                  <span
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "display": "block",
                        "fontSize": 14,
                        "muiPrepared": true,
                      }
                    }>
                    123
                  </span>
                </div>
              </div>
            </CardHeader>
            <CardActions>
              <div
                style={
                  Object {
                    "muiPrepared": true,
                    "padding": 8,
                    "position": "relative",
                  }
                }>
                <RaisedButton
                  className="start"
                  disabled={false}
                  fullWidth={false}
                  label="Start"
                  labelPosition="after"
                  onClick={[Function]}
                  primary={true}
                  secondary={false}
                  style={
                    Object {
                      "marginRight": 8,
                    }
                  }>
                  <Paper
                    circle={false}
                    className="start"
                    rounded={true}
                    style={
                      Object {
                        "display": "inline-block",
                        "marginRight": 8,
                        "minWidth": 88,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      }
                    }
                    transitionEnabled={true}
                    zDepth={1}>
                    <div
                      className="start"
                      style={
                        Object {
                          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                          "backgroundColor": "#ffffff",
                          "borderRadius": "2px",
                          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                                 0 1px 4px rgba(0, 0, 0, 0.12)",
                          "boxSizing": "border-box",
                          "color": "rgba(0, 0, 0, 0.87)",
                          "display": "inline-block",
                          "fontFamily": "Roboto, sans-serif",
                          "marginRight": 8,
                          "minWidth": 88,
                          "muiPrepared": true,
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        }
                      }>
                      <EnhancedButton
                        containerElement="button"
                        disabled={false}
                        focusRippleColor="#ffffff"
                        focusRippleOpacity={0.16}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onKeyboardFocus={[Function]}
                        onMouseDown={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        onTouchTap={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#00bcd4",
                            "borderRadius": 2,
                            "height": 36,
                            "lineHeight": "36px",
                            "padding": 0,
                            "position": "relative",
                            "textAlign": "center",
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                            "width": "100%",
                          }
                        }
                        tabIndex={0}
                        touchRippleColor="#ffffff"
                        touchRippleOpacity={0.16}
                        type="button">
                        <button
                          disabled={false}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}
                          onTouchTap={[Function]}
                          style={
                            Object {
                              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                              "backgroundColor": "#00bcd4",
                              "border": 10,
                              "borderRadius": 2,
                              "boxSizing": "border-box",
                              "cursor": "pointer",
                              "display": "inline-block",
                              "fontFamily": "Roboto, sans-serif",
                              "fontSize": "inherit",
                              "fontWeight": "inherit",
                              "height": 36,
                              "lineHeight": "36px",
                              "margin": 0,
                              "muiPrepared": true,
                              "outline": "none",
                              "padding": 0,
                              "position": "relative",
                              "textAlign": "center",
                              "textDecoration": "none",
                              "transform": "translate(0, 0)",
                              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                              "verticalAlign": null,
                              "width": "100%",
                            }
                          }
                          tabIndex={0}
                          type="button">
                          <TouchRipple
                            abortOnScroll={true}
                            color="#ffffff"
                            opacity={0.16}>
                            <div
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}>
                              <div
                                style={
                                  Object {
                                    "backgroundColor": false,
                                    "borderRadius": 2,
                                    "height": 36,
                                    "muiPrepared": true,
                                    "top": 0,
                                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                  }
                                }>
                                <span
                                  style={
                                    Object {
                                      "color": "#ffffff",
                                      "fontSize": 14,
                                      "fontWeight": 500,
                                      "letterSpacing": 0,
                                      "margin": 0,
                                      "muiPrepared": true,
                                      "opacity": 1,
                                      "paddingLeft": 16,
                                      "paddingRight": 16,
                                      "position": "relative",
                                      "textTransform": "uppercase",
                                      "userSelect": "none",
                                    }
                                  }>
                                  Start
                                </span>
                              </div>
                            </div>
                          </TouchRipple>
                        </button>
                      </EnhancedButton>
                    </div>
                  </Paper>
                </RaisedButton>
                <RaisedButton
                  className="stop"
                  disabled={true}
                  fullWidth={false}
                  label="Stop"
                  labelPosition="after"
                  onClick={[Function]}
                  primary={false}
                  secondary={false}
                  style={
                    Object {
                      "marginRight": 8,
                    }
                  }>
                  <Paper
                    circle={false}
                    className="stop"
                    rounded={true}
                    style={
                      Object {
                        "display": "inline-block",
                        "marginRight": 8,
                        "minWidth": 88,
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      }
                    }
                    transitionEnabled={true}
                    zDepth={0}>
                    <div
                      className="stop"
                      style={
                        Object {
                          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                          "backgroundColor": "#ffffff",
                          "borderRadius": "2px",
                          "boxShadow": undefined,
                          "boxSizing": "border-box",
                          "color": "rgba(0, 0, 0, 0.87)",
                          "display": "inline-block",
                          "fontFamily": "Roboto, sans-serif",
                          "marginRight": 8,
                          "minWidth": 88,
                          "muiPrepared": true,
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        }
                      }>
                      <EnhancedButton
                        containerElement="button"
                        disabled={true}
                        focusRippleColor="rgba(0, 0, 0, 0.3)"
                        focusRippleOpacity={0.1}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onKeyboardFocus={[Function]}
                        onMouseDown={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        onTouchTap={[Function]}
                        style={
                          Object {
                            "backgroundColor": "rgb(229, 229, 229)",
                            "borderRadius": 2,
                            "height": 36,
                            "lineHeight": "36px",
                            "padding": 0,
                            "position": "relative",
                            "textAlign": "center",
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                            "width": "100%",
                          }
                        }
                        tabIndex={0}
                        touchRippleColor="rgba(0, 0, 0, 0.3)"
                        touchRippleOpacity={0.1}
                        type="button">
                        <button
                          disabled={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}
                          onTouchTap={[Function]}
                          style={
                            Object {
                              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                              "backgroundColor": "rgb(229, 229, 229)",
                              "border": 10,
                              "borderRadius": 2,
                              "boxSizing": "border-box",
                              "cursor": "default",
                              "display": "inline-block",
                              "fontFamily": "Roboto, sans-serif",
                              "fontSize": "inherit",
                              "fontWeight": "inherit",
                              "height": 36,
                              "lineHeight": "36px",
                              "margin": 0,
                              "muiPrepared": true,
                              "outline": "none",
                              "padding": 0,
                              "position": "relative",
                              "textAlign": "center",
                              "textDecoration": "none",
                              "transform": "translate(0, 0)",
                              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                              "verticalAlign": null,
                              "width": "100%",
                            }
                          }
                          tabIndex={-1}
                          type="button">
                          <div
                            style={
                              Object {
                                "backgroundColor": false,
                                "borderRadius": 2,
                                "height": 36,
                                "muiPrepared": true,
                                "top": 0,
                                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                              }
                            }>
                            <span
                              style={
                                Object {
                                  "color": "rgba(0, 0, 0, 0.3)",
                                  "fontSize": 14,
                                  "fontWeight": 500,
                                  "letterSpacing": 0,
                                  "margin": 0,
                                  "muiPrepared": true,
                                  "opacity": 1,
                                  "paddingLeft": 16,
                                  "paddingRight": 16,
                                  "position": "relative",
                                  "textTransform": "uppercase",
                                  "userSelect": "none",
                                }
                              }>
                              Stop
                            </span>
                          </div>
                        </button>
                      </EnhancedButton>
                    </div>
                  </Paper>
                </RaisedButton>
              </div>
            </CardActions>
            <Snackbar
              autoHideDuration={4000}
              message=""
              onRequestClose={[Function]}
              open={false}
              style={
                Object {
                  "textAlign": "center",
                }
              }>
              <ClickAwayListener
                onClickAway={null}>
                <div
                  open={false}
                  style={
                    Object {
                      "bottom": 0,
                      "display": "flex",
                      "left": "50%",
                      "muiPrepared": true,
                      "position": "fixed",
                      "textAlign": "center",
                      "transform": "translate(-50%, 48px)",
                      "transition": "transform 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, visibility 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "visibility": "hidden",
                      "zIndex": 2900,
                    }
                  }>
                  <WithWidth
                    message=""
                    open={false}>
                    <EventListener
                      onResize={[Function]}
                      target="window">
                      <SnackbarBody
                        message=""
                        open={false}
                        width={3}>
                        <div
                          style={
                            Object {
                              "backgroundColor": "rgba(0, 0, 0, 0.87)",
                              "borderRadius": 2,
                              "flexGrow": 0,
                              "fontFamily": "Roboto, sans-serif",
                              "height": 48,
                              "lineHeight": "48px",
                              "maxWidth": 568,
                              "minWidth": 288,
                              "muiPrepared": true,
                              "padding": "0 24px",
                              "width": "auto",
                            }
                          }
                          width={3}>
                          <div
                            style={
                              Object {
                                "color": "#ffffff",
                                "fontSize": 14,
                                "muiPrepared": true,
                                "opacity": 0,
                                "transition": "opacity 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                              }
                            }>
                            <span />
                          </div>
                        </div>
                      </SnackbarBody>
                    </EventListener>
                  </WithWidth>
                </div>
              </ClickAwayListener>
            </Snackbar>
          </div>
        </div>
      </Paper>
    </Card>
  </DashboardItem>
</MuiThemeProvider>
`;
